internal/testing/cluster_test.go

```diff
@@ -263,6 +263,33 @@ func TestUploadDownload(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	// upload two files under /foo
+	file1 := make([]byte, rhpv2.SectorSize/12)
+	file2 := make([]byte, rhpv2.SectorSize/12)
+	frand.Read(file1)
+	frand.Read(file2)
+	if err := w.UploadObject(context.Background(), bytes.NewReader(file1), "foo/file1"); err != nil {
+		t.Fatal(err)
+	}
+	if err := w.UploadObject(context.Background(), bytes.NewReader(file2), "foo/file2"); err != nil {
+		t.Fatal(err)
+	}
+
+	// fetch entries with "file" prefix
+	_, entries, err := cluster.Bus.Object(context.Background(), "foo/", "file", 0, -1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(entries) != 2 {
+		t.Fatal("expected two entry to be returned", len(entries))
+	}
+
+	// fetch entries with "foo" prefix
+	_, _, err = cluster.Bus.Object(context.Background(), "foo/", "foo", 0, -1)
+	if errors.Is(err, api.ErrObjectNotFound) {
```

Comment:
I don't think this can ever be true? Because we receive the error through an API so `errors.Is` probably doesn't work.

Regardless I think that this should return a `nil` error and no entries right? Since `foo/` exists?

Reply:
Oops. I removed the 404 all together. We could return a 404 but we'd have to check for entries with no prefix and offset 0 every time we find no entries.